githubutil/githubutil.go

```diff
@@ -17,6 +17,21 @@ import (
 	"golang.org/x/oauth2"
 )
 
+// tokenSource implements oauth2.TokenSource for GitHub App authentication
+type tokenSource struct {
+	ClientID       string
+	InstallationID int64
+	PrivateKey     string
+}
+
+func (t *tokenSource) Token() (*oauth2.Token, error) {
+	token, err := GenerateInstallationToken(context.Background(), t.ClientID, t.InstallationID, t.PrivateKey)
+	if err != nil {
+		return nil, err
+	}
+	return &oauth2.Token{AccessToken: token, Expiry: time.Now().Add(15 * time.Minute)}, nil
```

Comment:
I'm wondering if it makes sense for us to set a custom expiration time rather than just setting the expiration time returned by the server in `GenerateInstallationToken`. The installation token expiration is set by the server (and returned in the `client.Apps.CreateInstallationToken` response). We are not revoking the token, so by using a shorter expiration we are only creating more live tokens than necessary, which is bad.

I would suggest updating `GenerateInstallationToken` to also return the token expiration, and use it in here.